Airside_Functions.cpp

```diff
@@ -0,0 +1,458 @@
+// care of by the Xbee communication directly
+/**************************************************************************************************/
+
+#include "Mavlink2_lib/common/mavlink.h"
+#include "Airside_Functions.hpp"
+
+mavlink_decoding_status_t Mavlink_airside_decoder(PIGO_Message_IDs_e* type, uint8_t incomingByte, uint8_t *telemetryData)
+{
+    int channel = MAVLINK_COMM_0; //mavlink default one channel
+    PIGO_Message_IDs_e decoded_message_type = MESSAGE_ID_NONE;
+    mavlink_decoding_status_t decoding_status = MAVLINK_DECODING_INCOMPLETE;
+
+    mavlink_status_t status;
+    memset(&status, 0x00, sizeof(mavlink_status_t));
+
+    mavlink_message_t decoded_msg;
+    memset(&decoded_msg, 0, sizeof(mavlink_message_t));
+
+    uint8_t message_received = mavlink_parse_char(channel, incomingByte, &decoded_msg, &status);
+
+    if (message_received)
+    {
+        if (telemetryData == NULL)
+        {
+            return MAVLINK_DECODING_FAIL;
+        }
+
+        switch(decoded_msg.msgid)
+        {
+            case MAVLINK_MSG_ID_GLOBAL_POSITION_INT: // ID for GLOBAL_POSITION_INT, 33
+                {
+                    mavlink_global_position_int_t global_position;
+                    memset(&global_position, 0x00, sizeof(mavlink_global_position_int_t));
+
+                    mavlink_msg_global_position_int_decode(&decoded_msg, &global_position);
+                    uint32_t warg_ID = global_position.time_boot_ms;
+
+                        if (warg_ID == MESSAGE_ID_GPS_LANDING_SPOT)
+                        {
+                            PIGO_GPS_LANDING_SPOT_t landing_spot;
+                            memset(&landing_spot, 0x00, sizeof(PIGO_GPS_LANDING_SPOT_t));
+
+                            landing_spot.latitude = global_position.lat;
+                            landing_spot.longitude = global_position.lon;
+                            landing_spot.altitude = global_position.alt;
+                            landing_spot.landingDirection = global_position.relative_alt;
+
+                            memcpy((void*) telemetryData, (void*) &landing_spot, sizeof(PIGO_GPS_LANDING_SPOT_t));
+
+                            decoding_status = MAVLINK_DECODING_OKAY;
+                        }
+
+                        else if (   warg_ID == MESSAGE_ID_WAYPOINTS ||
+                                    warg_ID == MESSAGE_ID_HOMEBASE)
+                        {
+                            PIGO_WAYPOINTS_t waypoints;
+                            memset(&waypoints, 0x00, sizeof(PIGO_WAYPOINTS_t));
+
+                            waypoints.latitude = global_position.lat;
+                            waypoints.longitude = global_position.lon;
+                            waypoints.altitude = global_position.alt;
+                            waypoints.turnRadius = global_position.relative_alt;
+                            waypoints.waypointType = global_position.hdg;
```

Comment:
How are we going to ensure that the waypointType will be stored in hdg? It feels like it would be stored in `vx` since it's the next parameter in the __mavlink_global_position_int_t struct. 

(Just curious)

Reply:
It's all taken care by the encoder, the encoder would take a struct for waypoint, and it will assign the values to the corresponding items under the Mavlink GPS struct. The reason why I chose to use `hdg` for waypointType is because the waypointType is `uint8_t`, and the hdg is the closest thing (`uint16_t`)